app/models/job/Job.scala

```diff
@@ -260,9 +260,9 @@ class JobDAO @Inject()(sqlClient: SqlClient)(implicit ec: ExecutionContext)
     for {
       _ <- assertUpdateAccess(id)
       _ <- run(q"""UPDATE webknossos.jobs
-             SET state = 'PENDING', retriedBySuperUser = true
+             SET state = ${JobState.PENDING}, manualState = NULL, retriedBySuperUser = true
              WHERE _id = $id
-             AND state = 'FAILURE'""".asUpdate)
+             AND state IN (${JobState.FAILURE}, ${JobState.CANCELLED})""".asUpdate)
```

Comment:
The logic here is a little off imo (that's why my testing did not succeed). As I understand it, you want to be able to restart a manually cancelled job, correct? In that case the WHERE statement will not catch the case of a manually cancelled state that was pending (`state = PENDING manualState = CANCELLED`). I adjusted the query accordingly to also be able to work in this case.

Reply:
Thanks, I had assumed that sending the manual state to CANCELLED would also set the state to CANCELLED, but obviously that is not the case.